src/pytorch_lightning/utilities/rank_zero.py

```diff
@@ -35,20 +37,22 @@ def wrapped_fn(*args: Any, **kwargs: Any) -> Optional[Any]:
     return wrapped_fn
 
 
-# TODO: this should be part of the cluster environment
-def _get_rank() -> int:
+def _get_rank(trainer: Optional["pl.Trainer"] = None) -> Optional[int]:
+    if trainer is not None:
+        return trainer.global_rank if trainer.world_size > 1 else None
```

Comment:
But returning None when world_size=1 is strange. For any single-device Trainer, rank should return 0 here. 
```suggestion
        return trainer.global_rank
```

Reply:
This is a quirk in the `EarlyStopping._log_info` implementation. I can move it to the `_log_info` impl if you prefer https://github.com/Lightning-AI/lightning/pull/14116/commits/f11c5de6648c915329271d191bd1ab338e55debf

Reply:
Or we can change the behaviour and update the tests that fail without it

Reply:
Ok, so this looks like it is just about the cosmetic decision of whether to prefix the rank or not on single-device execution. Either way is fine with me.